src/App.tsx

```diff
@@ -16,35 +24,84 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [visibleGoods, setVisibleGoods] = useState<string[]>(goodsFromServer);
+  const [sortOrder, setSortOrder] = useState<SortType>(SortType.None);
+
+  const sortAlphabetically = () => {
+    setVisibleGoods(
+      [...visibleGoods].sort((good1, good2) => good1.localeCompare(good2)),
+    );
+    setSortOrder(SortType.Alphabetical);
+  };
+
+  const sortByItemLength = () => {
+    setVisibleGoods([...visibleGoods].sort((a, b) => a.length - b.length));
+    setSortOrder(SortType.Length);
+  };
+
+  const reverseGoodsList = () => {
+    setVisibleGoods([...visibleGoods].reverse());
+    setSortOrder(SortType.Reverse);
+  };
```

Comment:
The `reverseGoodsList` function reverses the `visibleGoods` array without considering the current sort order. If the list was previously sorted, reversing it might not yield the expected results. Consider implementing a more intuitive approach, such as reversing based on the original order or the current sorted order.